src/param-builder.ts

```diff
@@ -160,7 +167,9 @@ class ParamBuilder {
     if (path && plain) mods.push('plain', path);
     else mods.push(encodeFilePath(path));
 
-    const res = mods.join('/');
+    const extension = options?.addExtension ? this.getExtention(path) : '';
```

Comment:
The extensions only work with encoded URL's, so we probably need to check that plain mode is not enabled